lib/pbench/test/unit/server/query_apis/conftest.py

```diff
@@ -29,17 +29,21 @@ def query_api(
         payload: Dict[str, Any],
         expected_index: str,
         expected_status: str,
+        authorization: bool = True,
         **kwargs,
     ) -> requests.Response:
         host = server_config.get("elasticsearch", "host")
         port = server_config.get("elasticsearch", "port")
         es_url = f"http://{host}:{port}{expected_index}{es_uri}"
         with responses.RequestsMock() as rsp:
             rsp.add(responses.POST, es_url, **kwargs)
+            if authorization:
+                headers = {"Authorization": "Bearer " + pbench_token}
+            else:
+                headers = {}
             response = client.post(
-                f"{server_config.rest_uri}{pbench_uri}", json=payload
+                f"{server_config.rest_uri}{pbench_uri}", headers=headers, json=payload,
```

Comment:
Ah, a little tricky one though, most of the request need an authentication token in the header since the request json we build has a valid "user" field in it. So I will have to use `pbench_token` fixture repetitively in lot of tests instead of using just once in `query_api`. Will that be a desirable?

Reply:
IIUC, the value of a text fixture is [cached by PyTest](https://docs.pytest.org/en/6.2.x/fixture.html#fixtures-can-be-requested-more-than-once-per-test-return-values-are-cached), so I don't think it should be a problem to use `pbench_token` repeatedly.

Making the default be the most-used value is appealing, but what is probably more important is to use the parameter in the way which makes it easiest to understand the tests.  For that, I'm inclined to suggest that omitting the parameter should imply omitting the value (i.e., the default value should be `{}`).  The fact that all the other invocations then have to supply the token is not necessarily a bad thing -- that makes it explicit what they are doing, which is generally good for understanding.